core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -360,6 +361,16 @@ Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
             Collection<ComputedStatistics> computedStatistics,
             List<TableHandle> sourceTableHandles);
 
+    /**
+     * Push update into connector
+     */
+    Optional<TableHandle> applyUpdate(Session session, TableHandle tableHandle, Map<String, Constant> assignments);
```

Comment:
What’s the key in the assignments map? If that’s meant to be the columns, it should be ColumnHandles. The SPI should not be dealing with column names at this point.

Reply:
@martint thx! refactored